### PR TITLE
Supabase 에러 메시지 [object Object] 마스킹 버그 수정

### PR DESCRIPTION
**Pull request by BumgeunSong**
## Summary
- Supabase `PostgrestError`가 plain object라 `String()` 변환 시 `"[object Object]"`로 표시되던 버그 수정
- `getErrorMessage()` 헬퍼 추가로 `.message` 프로퍼티를 안전하게 추출
- `captureQueryError`, `captureMutationError`, `addErrorContext` 3곳에 적용

## Context
- Sentry Issue ID: 7152355728
- 유저가 notifications 페이지에서 Supabase 권한 에러 발생 시, Sentry에 실제 에러 메시지 대신 `[object Object]`가 기록됨
- PostgrestError는 `{ message, details, hint, code }` 형태의 plain object로 `Error` 인스턴스가 아님

## Test plan
- [x] `getErrorMessage` 유닛 테스트 9개 추가 및 통과
- [x] 기존 테스트 481개 전체 통과 (5개 실패는 기존 Supabase env 미설정 이슈)

### CODE REVIEW

File: src/shared/lib/queryErrorTracking.ts [hunk 1]
```diff
@@ -9,6 +9,23 @@ import { getCurrentUserIdFromStorage } from '@/shared/utils/getCurrentUserId';
...
+/**
+ * Extract a meaningful message from any error-like value.
+ *
+ * Supabase PostgrestError is a plain object with a `message` property
+ * but is NOT an Error instance, so `String(error)` yields "[object Object]".
+ */
+export function getErrorMessage(error: unknown): string {
+  if (error instanceof Error) return error.message;
+  if (typeof error === 'object' && error !== null && 'message' in error) {
+    return String((error as { message: unknown }).message);
```
[the remark this answers is not in the document]
Comment (BumgeunSong) [reply src/shared/lib/queryErrorTracking.ts]: Fixed. Now extracts `message` first and checks its type — `string` returned directly, objects go through `JSON.stringify`, rest through `String()`. Added a test for the nested object edge case.
